sdk/communication/azure-communication-administration/azure/communication/administration/_phone_number_administration_client.py

```diff
@@ -440,37 +503,68 @@ def list_all_searches(
         )
 
     @distributed_trace
-    def cancel_search(
+    def cancel_reservation(
         self,
-        search_id,  # type: str
+        reservation_id,  # type: str
         **kwargs  # type: Any
     ):
         # type: (...) -> None
-        """Cancels the search. This means existing numbers in the search will be made available.
+        """Cancels the reservation. This means existing numbers in the reservation will be made available.
 
-        :param search_id: The search id to be canceled.
-        :type search_id: str
+        :param reservation_id: The reservation id to be canceled.
+        :type reservation_id: str
         :rtype: None
         """
         return self._phone_number_administration_client.phone_number_administration.cancel_search(
-            search_id,
+            search_id=reservation_id,
             **kwargs
         )
 
     @distributed_trace
-    def purchase_search(
-            self,
-            search_id,  # type: str
-            **kwargs  # type: Any
+    def begin_purchase_reservation(
+        self,
+        **kwargs  # type: Any
     ):
-        # type: (...) -> None
-        """Purchases the phone number search.
-
-        :param search_id: The search id to be purchased.
-        :type search_id: str
-        :rtype: None
+        # type: (...) -> LROPoller
+        """Begins purchase the reserved phone numbers of a phone number search.
+        Caller must provide either reservation_id, or continuation_token keywords to use the method.
+        If both body and continuation_token are specified, only continuation_token will be used to
```

Comment:
here too: `body`

Reply:
Yes that is a valuable point. We need to sync with other team members to make the args here consistent across languages.

Reply:
I have correct it. It should be `reservation_id`